addons/web_editor/tools.py

```diff
@@ -0,0 +1,136 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+import odoo
+
+from odoo import _
+from odoo.tools import image_process
+from odoo.modules.module import get_module_resource
+
+# To detect if we have a valid URL or not
+valid_url_regex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+# Regex for few of the widely used video hosting services
+player_regex = {
+    'youtube' : r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})(?:\S+)?$',
+    'vimeo' : r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*',
+    'dailymotion' : r'(?:.+dailymotion.com\/(video|hub|embed\/video|embed)|dai\.ly)\/([^_?]+)[^#]*(#video=([^_&]+))?',
+    'vine' : r'\/\/vine.co\/v\/(.[a-zA-Z0-9]*)',
```

Comment:
@JKE-be Vine is not operational anymore. Should we remove it?

Reply:
no, you can remove it if it doesn't works ;)
That will be not usefull from what I can read: "Twitter released the Vine Archive shortly after closing Vine, but, as of 2019, the Vine Archive is no longer supported."

So we can just drop this video provider